packages/hydrogen-react/src/load-script.tsx

```diff
@@ -62,23 +62,17 @@ export function useLoadScript(
   options?: LoadScriptParams[1],
 ): ScriptState {
   const [status, setStatus] = useState<ScriptState>('loading');
-  const stringifiedOptions = JSON.stringify(options);
 
-  useEffect(() => {
-    async function loadScriptWrapper(): Promise<void> {
-      try {
-        setStatus('loading');
-        await loadScript(url, options);
-        setStatus('done');
-      } catch (error) {
-        setStatus('error');
-      }
-    }
-
-    loadScriptWrapper().catch(() => {
-      setStatus('error');
-    });
-  }, [url, stringifiedOptions, options]);
+  useEffect(
+    () => {
+      loadScript(url, options)
+        .then(() => setStatus('done'))
+        .catch(() => setStatus('error'));
+    },
+    // Ignore options changes since it won't trigger a new load.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [url],
+  );
```

Comment:
The `options` object is only used the first time. The resulting promise is cached based on `url`, so even if they change `options` we are not refreshing the script loaded after the first time. That's why I'm not adding it to the useEffect dependencies here.